File: easy/997. Find the Town Judge/solution.rs
impl Solution {
    pub fn find_judge(n: i32, trust: Vec<Vec<i32>>) -> i32 {
        let n = n as usize;
        let mut indeg = vec![0i32; n];
        let mut outdeg = vec![0i32; n];

        for t in trust {
            let a = (t[0] - 1) as usize;
            let b = (t[1] - 1) as usize;
            outdeg[a] += 1;
            indeg[b] += 1;
        }

        for i in 0..n {
            if indeg[i] == (n as i32 - 1) && outdeg[i] == 0 {
                return (i as i32) + 1;
            }
        }

        -1
    }
}

File: easy/997. Find the Town Judge/solution.c
#include <stdio.h>
#include <stdlib.h>

/*
    Find the town judge.

    Idea:
    - The judge trusts nobody  -> outdegree = 0
    - Everybody else trusts the judge -> indegree = n - 1
    - We count indegree and outdegree for each person.
*/

int findJudge(int n, int** trust, int trustSize, int* trustColSize) {
    // Allocate memory for indegree and outdegree arrays
    int* indeg = (int*)calloc(n, sizeof(int));
    int* outdeg = (int*)calloc(n, sizeof(int));

    // Process trust relationships
    for (int i = 0; i < trustSize; i++) {
        int a = trust[i][0] - 1;  // convert to 0-based index
        int b = trust[i][1] - 1;

        outdeg[a]++;  // a trusts someone
        indeg[b]++;   // b is trusted by someone
    }

    // Check judge conditions
    for (int i = 0; i < n; i++) {
        if (indeg[i] == n - 1 && outdeg[i] == 0) {
            free(indeg);
            free(outdeg);
            return i + 1;  // convert back to 1-based index
        }
    }

    free(indeg);
    free(outdeg);
    return -1;  // no judge found
}